xtable-api/src/main/java/org/apache/xtable/spi/sync/ConversionTarget.java

```diff
@@ -76,8 +76,9 @@ public interface ConversionTarget {
    * Starts the sync and performs any initialization required
    *
    * @param table the table that will be synced
+   * @param sourceIdentifier
    */
-  void beginSync(InternalTable table);
+  void beginSync(InternalTable table, String sourceIdentifier);
```

Comment:
I am wondering if it makes sense to attach the source format as part of the identifier. Then we can also have some metadata about what the writer format was.

Reply:
> I am wondering if it makes sense to attach the source format as part of the identifier. Then we can also have some metadata about what the writer format was.

Good idea. I create a `SourceMetadata` as a commit-level information attached in target table commit so that It would be much easier if we want add more information in the future